dot/node_integration_test.go

```diff
@@ -0,0 +1,397 @@
+	cfg.Core.GrandpaAuthority = false
+	cfg.Init.Genesis = genPath
+
+	gen, err := genesis.NewGenesisFromJSONRaw(genPath)
+	require.NoError(t, err)
+
+	err = InitNode(cfg)
+	require.NoError(t, err)
+
+	ks := keystore.NewGlobalKeystore()
+	ed25519Keyring, _ := keystore.NewEd25519Keyring()
+	ks.Gran.Insert(ed25519Keyring.Alice())
+	sr25519Keyring, _ := keystore.NewSr25519Keyring()
+	ks.Babe.Insert(sr25519Keyring.Alice())
+	node, err := NewNode(cfg, ks)
+	require.NoError(t, err)
+
+	if reflect.TypeOf(node) != reflect.TypeOf(&Node{}) {
+		t.Fatalf("failed to return correct type: got %v expected %v", reflect.TypeOf(node), reflect.TypeOf(&Node{}))
+	}
+
+	expected := &trie.Trie{}
+	err = expected.LoadFromMap(gen.GenesisFields().Raw["top"])
+	require.NoError(t, err)
+
+	expectedRoot, err := expected.Hash()
+	require.NoError(t, err)
+
+	mgr := node.Services.Get(&core.Service{})
+
+	var coreSrvc *core.Service
+	var ok bool
+
+	if coreSrvc, ok = mgr.(*core.Service); !ok {
+		t.Fatal("could not find core service")
+	}
+	require.NotNil(t, coreSrvc)
+
+	stateRoot, err := coreSrvc.StorageRoot()
+	require.NoError(t, err)
+	require.Equal(t, expectedRoot, stateRoot)
+}
+
+// balanceKey returns the storage trie key for the balance of the account with the given public key
+func balanceKey(t *testing.T, key [32]byte) []byte {
+	accKey := append([]byte("balance:"), key[:]...)
+	hash, err := common.Blake2bHash(accKey)
+	require.NoError(t, err)
+	return hash[:]
+}
+
+func TestInitNode_LoadBalances(t *testing.T) {
+	cfg := NewTestConfig(t)
+	require.NotNil(t, cfg)
+
+	genPath := NewTestGenesisAndRuntime(t)
+	require.NotNil(t, genPath)
+
+	defer utils.RemoveTestDir(t)
+
+	cfg.Core.Roles = types.FullNodeRole
+	cfg.Core.BabeAuthority = false
+	cfg.Core.GrandpaAuthority = false
+	cfg.Init.Genesis = genPath
+
+	err := InitNode(cfg)
+	require.NoError(t, err)
+
+	ks := keystore.NewGlobalKeystore()
+	ed25519Keyring, _ := keystore.NewEd25519Keyring()
+	ks.Gran.Insert(ed25519Keyring.Alice())
+
+	node, err := NewNode(cfg, ks)
+	require.NoError(t, err)
+
+	if reflect.TypeOf(node) != reflect.TypeOf(&Node{}) {
+		t.Fatalf("failed to return correct type: got %v expected %v", reflect.TypeOf(node), reflect.TypeOf(&Node{}))
+	}
+
+	mgr := node.Services.Get(&state.Service{})
+
+	var stateSrv *state.Service
+	var ok bool
+
+	if stateSrv, ok = mgr.(*state.Service); !ok {
+		t.Fatal("could not find core service")
+	}
+	require.NotNil(t, stateSrv)
+
+	kr, _ := keystore.NewSr25519Keyring()
+	alice := kr.Alice().Public().(*sr25519.PublicKey).AsBytes()
+
+	bal, err := stateSrv.Storage.GetStorage(nil, balanceKey(t, alice))
+	require.NoError(t, err)
+
+	genbal := "0x0000000000000001"
+	expected, _ := common.HexToBytes(genbal)
+	require.Equal(t, expected, bal)
+}
+
+func TestNode_StopFunc(t *testing.T) {
+	testvar := "before"
+
+	node := &Node{
+		Services: &services.ServiceRegistry{},
+		wg:       sync.WaitGroup{},
+	}
+	node.wg.Add(1)
+
+	node.Stop()
+	require.Equal(t, testvar, "after")
```

Comment:
Wait I'm confused how is `testVar` changed here?

Reply:
This test in no longer needed, so it's been removed.